onnxruntime/python/tools/quantization/calibrate.py

```diff
@@ -345,25 +345,31 @@ def calibrate(model_path,
               op_types=['Conv', 'MatMul'],
               black_nodes=[],
               white_nodes=[],
-              augmented_model_path='augmented_model.onnx'):
+              augmented_model_path='augmented_model.onnx',
+              providers=["CPUExecutionProvider"],
+              tensorrt_calibration=False):
     '''
-        Given an onnx model, augment and run the augmented model on calibration data set, aggregate and calculate the quantization parameters.
+    Given an onnx model, augment and run the augmented model on calibration data set, aggregate and calculate the quantization parameters.
     :param model_path: ONNX model to calibrate
     :param data_reader: user implemented object to read in and preprocess calibration dataset based on CalibrationDataReader interface
     :param op_types: operator types to be calibrated and quantized, default = 'Conv,MatMul'
     :param black_nodes: operator names that should not be quantized, default = ''
     :param white_nodes: operator names that force to be quantized, default = ''
     :param augmented_model_path: save augmented_model to this path
+    :param providers: execution providers to run calibration
+    :tensorrt_calibration: TensorRT calibration. Quantization parameter calculation will be skipped 
     '''
     #1. initialize a calibrater
     calibrater = ONNXCalibrater(model_path, data_reader, op_types, black_nodes, white_nodes, augmented_model_path)
     #2. augment
-    augmented_model = calibrater.augment_graph()
+    augmented_model = calibrater.augment_graph(augment_all_ops=True if tensorrt_calibration else False)
```

Comment:
ideally we shouldn't have logic like this.
the input argument has op_types  
it specifies the ops to augment, we should use that directly instead of having another argument to bypass it.

Reply:
For TensorRT, we use tensor data type rather than op type to decide if augmentation is needed or not. All FP32 type tensors will be augmented.

Reply:
yes, but we should generically extend the method to support data type rather than optype. (because it could be applied to other backends as well)

Reply:
I think we need to support both op type and data type. For ORT CPU/CUDA EP, we know what INT8 onnx ops are supported, so op type is more precise for calibration. While for TRT (or other vendor's EP), we may not know exact op type it supports, then we use data type to get best INT8 calibration coverage.

Reply:
I think we need to support both data type and op type. For CPU/CUDA EP, we know what onnx INT8 ops are supported, so op type is better for calibration. While for TRT (or other vendor's EPs), we may not know what INT8 ops it supports, then data type is better to get more calibration coverage.

Reply:
yes, that's what i am saying. extend it to support either case.